diesel/src/query_dsl/distinct_dsl.rs

```diff
@@ -46,3 +47,55 @@ where
         self.as_query().distinct()
     }
 }
+
+/// Adds the `DISTINCT ON` clause to a query.
+///
+/// # Example
+///
+/// ```rust
+///
+/// # #[macro_use] extern crate diesel;
+/// # include!("../doctest_setup.rs");
+/// #
+/// # table! {
+/// #     users {
+/// #         id -> Integer,
+/// #         name -> VarChar,
+/// #     }
+/// # }
+/// #
+/// # fn main() {
+/// #     use self::animals::dsl::*;
+/// #     let connection = establish_connection();
+/// #     connection.execute("DELETE FROM animals").unwrap();
+/// connection.execute("INSERT INTO animals (species, name, legs) VALUES ('dog', 'Jack', 4), ('dog', Null, 4), ('spider', Null, 8)")
+///     .unwrap();
+/// let all_animals = animals.select((species, name, legs)).load(&connection);
+/// let distinct_animals = animals.select((species, name, legs)).distinct_on(species).load(&connection);
+///
+/// assert_eq!(Ok(vec![(String::from("dog"), Some(String::from("Jack")), 4),
+///                    (String::from("dog"), None, 4),
+///                    (String::from("spider"), None, 8)]), all_animals);
+/// assert_eq!(Ok(vec![(String::from("dog"), Some(String::from("Jack")), 4),
+///                    (String::from("spider"), None, 8)]), distinct_animals);
```

Comment:
This is hard to parse for me. Maybe swap the params so the var is first? Or make a small `Animal::new("dog", Some("Jack"), 4)` builder (i.e., `fn new<S: Into<String>, N: Into<String>>(species: S, name: Option<N>, legs: u32) { … }` to reduce the `String::from` noise)